### PR TITLE
update r-base and pins

### PR DESCRIPTION
**Pull request by ocefpaf**
@bgruening can you take a look at this one?

### CODE REVIEW

**Comment by bgruening** [comment]
@ocefpaf I'm not convinced we need the `3.4.2` release. Do you have any reasons? Compat with `default`? This would mean a lot of rebuilds and a lot of CI time, that is probably better spent elsewhere.

**Comment by ocefpaf** [comment]
> @ocefpaf I'm not convinced we need the `3.4.2` release. Do you have any reasons? Compat with `default`? This would mean a lot of rebuilds and a lot of CI time, that is probably better spent elsewhere.

We can postpone the `3.4.2`, no problem. But we do need to update the pinning otherwise we are risking some incompatibilities as we update other packages.

**Comment by bgruening** [comment]
Ok, lets use the older version.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ source:
     - 0013-javareconf-do-not-fail-on-compile-fail.patch
 
 build:
-  number: 2
+  number: 0
```
Comment (jdblischak) [inline recipe/meta.yaml]: @bgruening Shouldn't this be bumped to build number 3 if the plan is to stick with R 3.4.1?

Comment (bgruening) [reply recipe/meta.yaml]: Oh, yes good catch.

**Comment by jdblischak** [comment]
Is the plan to also attempt to update the pins for R 3.3.2?

**Comment by bgruening** [comment]
I don't think so, but we could do this in a branch.

**Comment by jdblischak** [comment]
> I don't think so, but we could do this in a branch.

Right, I was referring to the 3.3.2 branch. As long as the matrix build includes both 3.3.2 and 3.4.1, I feel like we need to commit to maintaining 3.3.2. Once that gets too burdensome, we should stop actively building new recipes for 3.3.2.

https://github.com/conda-forge/r-base-feedstock/tree/3.3.2

Am I thinking about this correctly? At some point we will have to stop supporting 3.3.2, at which point we'll need to change the build matrix used by staged-recipes and conda-smithy.

**Comment by bgruening** [comment]
Yes, I'm all in to update this branch as well. Lets see if this PR passes and then update the other branch as well.

**Comment by jakirkham** [comment]
> Am I thinking about this correctly? At some point we will have to stop supporting 3.3.2, at which point we'll need to change the build matrix used by staged-recipes and conda-smithy.

Hopefully we will be on `conda-build` 3 before then. By which point, the change will only need to be made in [this config file]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml ).

**Comment by bgruening** [comment]
@ocefpaf ready to be merged?

**Comment by ocefpaf** [comment]
Apologies @bgruening, lost track of this one. Merging!

**Comment by jdblischak** [comment]
What's the plan for updating the pins for the 3.3.2 branch? I'd be willing to submit the PR if you two are busy. I assume the pins are updated using the script [pin_the_slow_way.py](https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/pin_the_slow_way.py), right?

**Comment by bgruening** [comment]
Please go ahead. This is the correct script, but you can also just take the pinning from this PR.

**Pull request merged**